### PR TITLE
Replace golang/glog with kubevirt/client-go/log in virtctl

### LINKED ISSUE #11568: virtctl: use client-go/log
**Is your feature request related to a problem? Please describe**:
A clear and concise description of what the problem is.

We have a mix of components in virtctl that are using either `golang/glog` or `kubevirt.io/client-go/log`. 
For consistency, we should use the same logging module and I'd recommend our `client-go/log` as, if we need, we can change it latter to affect all components that use it.

I consider this a sub-task of https://github.com/kubevirt/kubevirt/issues/5932

**Describe the solution you'd like**:
- All `virtctl` modules should be using `client-go/log`
- They should all call `InitializeLogging` as well, to provide a reasonable component name.

### PR DESCRIPTION
**Pull request by harsh-vardhan-ande**
### What this PR does
Before this PR: Currently both golang/glog and kubevirt/client-go/log libraries are being used across the repo which should not happen. Only one of these should be used.

After this PR:
In the virtctl package, usages of golang/glog have been replaced with kubevirt/client-go/log as the latter provides better functionality and future compatibility.

Fixes #11568

### Why we need it and why it was done in this way
The following tradeoffs were made:
It is more maintainable to use only one logging package across the code hence glog was removed. 

The following alternatives were considered:

Links to places where the discussion took place: 

### Special notes for your reviewer

`log.InitializeLogging` was done in appropriate places where the library was used. However, I'm not entirely sure on its correct usage and couldn't find much documentation on it online. Please let me know if it has been used correctly or not.

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Comment by alicefr** [comment]
@harsh-vardhan-ande please adjust the commit title and body. The title should be short and the body should contains a brief explanation. I would suggest you to split the first sentence with simply "log: replace of golang/glog with kubevirt/client-go/log " and the rest as commit body

**Comment by alicefr** [comment]
The changes look to me. I'm wondering if the should provide a constant when we initialize the logger in multiple instances of the same command instead of using hardcoded strings like for portforward and ssh. @0xFelix @victortoso wdyt?

File: pkg/virtctl/portforward/tcp.go [hunk 1]
```diff
@@ -4,10 +4,11 @@ import (
 	"io"
 	"net"
 
-	"github.com/golang/glog"
+	"kubevirt.io/client-go/log"
 )
 
 func (p *portForwarder) startForwardingTCP(address *net.IPAddr, port forwardedPort) error {
+	log.InitializeLogging("portforward")
```
Comment (0xFelix) [inline pkg/virtctl/portforward/tcp.go:11]: Is it necessary to initialize the logger multiple times?

**Review by 0xFelix — COMMENTED** [review]
Is it really required to initialize the logging multiple times per package? I would have thought initializing it once per entrypoint should be enough? If it is really necessary then a constant would be nice.

File: pkg/virtctl/portforward/tcp.go (hunk 1 above)
Comment (harsh-vardhan-ande) [reply pkg/virtctl/portforward/tcp.go]: It shouldn't be needed. Have made changes to initialize only once per package in the entrypoint function in the package.

**Comment by harsh-vardhan-ande** [comment]
> @harsh-vardhan-ande please adjust the commit title and body. The title should be short and the body should contains a brief explanation. I would suggest you to split the first sentence with simply "log: replace of golang/glog with kubevirt/client-go/log " and the rest as commit body

Updated both my PRs according to this suggestion.

**Review by 0xFelix — APPROVED** [review]
Thanks, looks good to me!

/lgtm

**Comment by harsh-vardhan-ande** [comment]
@0xFelix Thanks for the review. One test is failing - `pull-kubevirt-conformance-arm64`. Following is the error message
```
failed to get image info: failed to invoke qemu-img: exit status 2: 'fatal error: runtime: cannot allocate memory
```
The test was trying to run a command using the `virt-chroot` binary but the command failed. It looks like the pod may have gone OOM and thats why it failed. 
Can we please retry this test?

**Comment by alicefr** [comment]
> @0xFelix Thanks for the review. One test is failing - `pull-kubevirt-conformance-arm64`. Following is the error message
> 
> ```
> failed to get image info: failed to invoke qemu-img: exit status 2: 'fatal error: runtime: cannot allocate memory
> ```
> 
> The test was trying to run a command using the `virt-chroot` binary but the command failed. It looks like the pod may have gone OOM and thats why it failed. Can we please retry this test?

The arm lanes aren't mandatory

**Comment by alicefr** [comment]
/approve
Looks good, thanks!

**Comment by 0xFelix** [comment]
This change breaks setting the logging verbosity for users. Only the default log verbosity can still be used.

Before this PR was merged it was possible to provide the arg `-v 3` to virtctl to set the verbosity to 3. Because the flag is no longer registered virtctl now fails with `unknown shorthand flag: 'v' in -v`.

@harsh-ande Can you work on a PR to fix this?

**Comment by 0xFelix** [comment]
Created https://github.com/kubevirt/kubevirt/issues/12779 for this regression.

**Pull request merged**